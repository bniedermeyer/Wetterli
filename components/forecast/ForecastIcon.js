import React from 'react';
import { View, Image, StyleSheet } from 'react-native';

export default function ForecastIcon(props) {
  if (!props.icon) {
    return null;
  }
  const images = {
    'clear-day': require('../../assets/images/weather/clear-day.png'),
    'clear-night': require('../../assets/images/weather/clear-night.png'),
    cloudy: require('../../assets/images/weather/cloudy.png'),
    fog: require('../../assets/images/weather/fog.png'),
    'partly-cloudy-day': require('../../assets/images/weather/partly-cloudy-day.png'),
    'partly-cloudy-night': require('../../assets/images/weather/partly-cloudy-night.png'),
    rain: require('../../assets/images/weather/rain.png'),
    sleet: require('../../assets/images/weather/sleet.png'),
    snow: require('../../assets/images/weather/snow.png'),
    thunderstorm: require('../../assets/images/weather/thunderstorm.png'),
    wind: require('../../assets/images/weather/wind.png')
  };

  return <Image source={images[props.icon]} style={styles.weatherImage} />;
}

const styles = StyleSheet.create({
  weatherImage: {
    width: 200,
    height: 200,
    resizeMode: 'contain',
    marginTop: 3,
    marginLeft: -10
  }
});
