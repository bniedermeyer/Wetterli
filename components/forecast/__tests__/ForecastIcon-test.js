import React from 'react';
import renderer from 'react-test-renderer';

import ForecastIcon from '../ForecastIcon';

it(`renders correctly`, () => {
  const tree = renderer.create(<ForecastIcon icon={'clear-day'} />).toJSON();

  expect(tree).toMatchSnapshot();
});
