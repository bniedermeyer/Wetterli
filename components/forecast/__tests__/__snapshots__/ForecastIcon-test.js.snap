// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Image
  source={1}
  style={
    Object {
      "height": 200,
      "marginLeft": -10,
      "marginTop": 3,
      "resizeMode": "contain",
      "width": 200,
    }
  }
/>
`;
